proxy/plugin/solana_rest_api.py

```
@@ -83,11 +82,19 @@ def eth_gasPrice(self):
         return 0
 
     def eth_estimateGas(self, param):
-        if param.get('to'):
-            result = call_emulated(param['to'], param['from'], param['data'])
+        try:
+            if not param['data']:
+                raise Exception("Missing data in eth_estimateGas param")
+            caller_id = param['from']
+            contract_id = param['to'] if 'to' in param else "deploy"
+            data = param['data'] if 'data' in param else ""
+            value = param['value'] if 'value' in param else ""
+            debug = param['debug'] if 'debug' in param else ""
+            result = call_emulated(contract_id, caller_id, data, value, debug)
             return result['used_gas']
-        else:
-            return 9999998
+        except Exception as err:
+            logger.debug("Exception on eth_estimateGas: %s", err)
+            return 999999999
```

rethrow an exception instead of returning an arbitrary value

---

Fixed